### ISSUE TITLE
Server-Side Rendering CSS Delay with @ant-design/nextjs-registry in Next.js

### ISSUE THREAD

**Issue by hamedyosefian** [ant-design/ant-design#47122]
### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/antd-reproduction-template-forked-jyh2k9)

### Steps to reproduce

Install and configure @ant-design/nextjs-registry in a Next.js app.
Implement Ant Design components in the application.
Run the application and observe the initial page load.

### What is expected?

The expectation is for Ant Design styles to be properly applied server-side, ensuring that the page is rendered with the correct styling on initial load without any delay.

### What is actually happening?

On the initial page load, the page appears without any Ant Design styles applied. The correct styling is only applied after a noticeable delay, leading to a flickering effect and a poor user experience. This behavior suggests an issue with server-side rendering of CSS when using @ant-design/nextjs-registry.

| Environment | Info |
| --- | --- |
| antd | 5.12.8 |
| React | 18 |
| System | macos |
| Browser | chrome |

**Comment by MadCcc** [comment]
Please provide reproduction and don't open duplicated issues.

**Comment by hamedyosefian** [comment]
On the initial page load, the page appears without any Ant Design styles applied. The correct styling is only applied after a noticeable delay, leading to a flickering effect and a poor user experience. This behavior suggests an issue with server-side rendering of CSS when using @ant-design/nextjs-registry.
When use await for server fetch data has this problem
i test it and when loading.jsx file this problem and when remove that file everthing is ok
and layout code is this :

<html lang='fa' dir='rtl'>
      <body id='app'>
        <AntdRegistry>
          <AntdConfigProvider
            direction='rtl'
            locale={faIR}
            theme={{
              token: {
                fontFamily: 'IRANYekanX',
                colorPrimary: '#374AFE',
              },
            }}
          >
            <AntdApp>
              <SWRProvider>
                <main>{children}</main>
              </SWRProvider>
            </AntdApp>
          </AntdConfigProvider>
        </AntdRegistry>
      </body>
    </html>
This code has delay:

import TestCompon from '@/TestCompon'

const getPosts = async () => {
  const data = await fetch('https://jsonplaceholder.typicode.com/posts')
  const posts = await data.json()
  return posts
}
export default async function Home() {
  const posts = await getPosts()
  console.log(posts)
  return (
    <div>
      <TestCompon />
    </div>
  )
}
This code is ok:

import TestCompon from '@/TestCompon'

const getPosts = async () => {
  const data = await fetch('https://jsonplaceholder.typicode.com/posts')
  const posts = await data.json()
  return posts
}
export default function Home() {
  const posts = getPosts()
  console.log(posts)
  return (
    <div>
      <TestCompon />
    </div>
  )
}

**Comment by MadCcc** [comment]
Duplicate of https://github.com/ant-design/ant-design/issues/47104

**Comment by kishoreGleniseOfficial** [comment]
> > Duplicate of #47104 , trace it in #47104
> 
> now how to fix this bug ??

Hey @hamedyosefian , I will drop Image links so I could confirm we both are facing the same issue.

initial rendor - https://i.stack.imgur.com/O1YNH.png

After a reload/refresh - https://i.stack.imgur.com/3mp24.png

What I am facing is that during an initial rendor when giving npm start after a build, the buttons appear in page with no style. Only after a page reload/refresh the styling of the button works. This is annoying. Please confirm whether this is what you are facing too ?

**Comment by hamedyosefian** [comment]
> > > Duplicate of #47104 , trace it in #47104
> > 
> > 
> > now how to fix this bug ??
> 
> Hey @hamedyosefian , I will drop Image links so I could confirm we both are facing the same issue.
> 
> initial rendor - https://i.stack.imgur.com/O1YNH.png
> 
> After a reload/refresh - https://i.stack.imgur.com/3mp24.png
> 
> What I am facing is that, during an initial rendor when giving npm start after a build, the buttons appear in page with no style. Only after a page reload/refresh the styling of the button works. This is annoying. Please confirm whether this is what you are facing too ? This happens in development too for me

Hey kishore
no my problem not this 
for me show without css for one second and after that load css
and each time refresh this problem is still

**Comment by abolfazl-zare** [comment]
hi
I'm experiencing the exact same issue as hamedyosefian

**Comment by kishoreGleniseOfficial** [comment]
> hi I'm experiencing the exact same issue as hamedyosefian

Use `@ant-design/cssinjs`. See documentation to set up in Next.js if you are using it.
[](https://ant.design/docs/react/use-with-next#using-pages-router)
